client/asset/btc/spv.go

```diff
@@ -1997,9 +1997,7 @@ func (w *walletExtender) signTransaction(tx *wire.MsgTx) error {
 		txIn.SignatureScript = nil
 		txIn.Witness = nil
 	}
-	return walletdb.View(w.Database(), func(dbtx walletdb.ReadTx) error { // what is the db tx for?
```

Comment:
I added the comment in the Electrum PR, but didn't investigate further.  I think an early draft of native SPV BTC started with a different approach, but the `secretSource` now manages the DB accesses (calls to `btcwallet/wallet.(*Wallet).AddressInfo`).